### PR TITLE
Add analytics events to amp-lightbox-gallery

### LINKED ISSUE #16651: Analytics tracking for amp-lightbox-gallery
Feature request:

It'd be great to have tracking system in amp-lightbox-gallery
https://www.ampproject.org/docs/reference/components/amp-lightbox-gallery

This way we could track all different navigation levels that the amp-lightbox-gallery presents giving us an overview of the use of them in our amp pages.

### PR DESCRIPTION
**Pull request by nainar**
Events added:
- lightboxOpened
- thumbnailsViewToggled
- controlsToggled
- descriptionOverflowToggled

Didn't implement slideChange as that is exposed through `<amp-carousel>`.

Fixes #16651

### CODE REVIEW

File: examples/accordion-events.amp.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<html ⚡>
```
Comment (cathyxz) [inline examples/accordion-events.amp.html]: This file probably doesn't belong in this PR. =)

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 2]
```diff
@@ -456,13 +458,16 @@ export class AmpLightboxGallery extends AMP.BaseElement {
         this.measureMutateElement(measureOverflowState, mutateOverflowState);
       });
     }
+    triggerAnalyticsEvent(this.element, 'descriptionToggled', {});
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: What is description toggled supposed to mean? `updateDescriptionBox` gets called every time you swipe to a new slide in order to update the contents of the description. It isn't a user action. The user can tap once to bring up or make all the controls disappear (gallery/close buttons, left/right arrows, AND the description). You can fire that in `onToggleControls_`, but I think this event doesn't really make sense. 
Propose to either: 1. remove this event, or 2. expose 'controlsToggled' instead.

File: extensions/amp-lightbox-gallery/0.1/test/integration/test-amp-lightbox-gallery.js [hunk 3]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/test/integration/test-amp-lightbox-gallery.js]: You'll get a merge conflict from https://github.com/ampproject/amphtml/pull/16824 here because this test suite is super flaky on Windows. Sadness, but super thanks for writing tests! I'll take a look and see what we can do to fix the Windows thing in a bit.

File: examples/article-with-lightbox-2.0-gallery.amp.html [hunk 4]
```diff
@@ -340,6 +340,41 @@
...
+          "trackDescriptionExpanded": {
+            "on": "descriptionExpanded",
+            "request": "descExpand"
+          }
+        }
+      }
+      </script>
+      </amp-analytics>
   <header id="top-bar">
     <button class="menu" title="Toggle Sidebar">
```
Comment (cathyxz) [inline examples/article-with-lightbox-2.0-gallery.amp.html]: Can we also add these analytics events in documentation? Preferably include some portion of your working example code here too. =)

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 5]
```diff
@@ -456,13 +458,16 @@ export class AmpLightboxGallery extends AMP.BaseElement {
...
+
   }
 
   /**
    * Toggle the overflow state of description box
    * @private
    */
   toggleDescriptionOverflow_() {
+    triggerAnalyticsEvent(this.element, 'descriptionExpanded', {});
```
[the remark this answers is not in the document]
Comment (nainar) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: I think it's more useful tracking any interaction. Changing the event name as you suggested.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js (hunk 2 above)
Comment (nainar) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: Let's expose `controlsToggled` cause I think that is worth tracking. Made the change in the latest commit.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 6]
```diff
@@ -309,6 +310,7 @@ export class AmpLightboxGallery extends AMP.BaseElement {
    * @private
    */
   findOrBuildCarousel_(lightboxGroupId) {
+    triggerAnalyticsEvent(this.element, 'lightboxOpened', {});
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: I don't know how I missed this earlier, but could you put this in `open_()`? There's technically nothing wrong with this, but logically I think you want to trigger this in the last `then` block of the `open_` function, along with `showControls_`.
